### PR TITLE
fix: Join acceptance tests

### PR DESCRIPTION
**Pull request by esuc4se**
## Context
We need to acceptance test for remote join & external join.

## Objective
- Add remote join & external join accesptance test
- Refactor trigger.js, world.js

## License
I confirm that this contribution is made under a BSD license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: features/step_definitions/trigger.js [hunk 1]
```diff
@@ -14,148 +15,227 @@ Before(
...
-        this.builds = null;
+
+        this.buildId = null;
+        this.pipelines = {};
     }
 );
 
 Given(
-    /^an existing pipeline on branch "(.*)" with the workflow jobs:$/,
+    /^an existing pipeline on branch "([^"]*)" with the trigger jobs:$/,
```
Comment (kkisic) [inline features/step_definitions/trigger.js:32]:
```suggestion
    /^an existing pipeline on branch "([^"]*)" with the workflow jobs:$/,
```
I feel like this should remain a `workflow`.

File: features/step_definitions/trigger.js [hunk 2]
```diff
@@ -14,148 +15,227 @@ Before(
...
 Then(
-    /^the "(.*)" job on branch "(.*)" is not triggered$/,
+    /^the "([^"]*)" job is triggered from "([^"]*)" on branch "([^"]*)" and "([^"]*)" on branch "([^"]*)"$/,
     {
         timeout: TIMEOUT
     },
-    function step(jobName, branch) {
-        let pipelineId = this.pipelineIdA;
+    async function step(joinJobName, parentJobName, parentBranchName, externalJobName, externalBranchName) {
+        // parent pipeline
```
Comment (yuichi10) [inline features/step_definitions/trigger.js:202]: I think it's obvious. We can delete this comment.

File: features/step_definitions/trigger.js [hunk 3]
```diff
@@ -14,148 +15,227 @@ Before(
...
-                let result = buildData.body || [];
+        const parentJob = parentPipeline.jobs.find(j => j.name === parentJobName);
+        const parentBuild = parentBuilds.find(b => b.jobId === parentJob.id);
 
-                result = result.filter(item => item.sha === this.sha);
+        Assert.oneOf(parentBuild.id, joinBuild.parentBuildId);
 
-                Assert.equal(result.length, 0, 'Unexpected job was triggered.');
-            });
+        // external pipeline
```
Comment (yuichi10) [inline features/step_definitions/trigger.js:220]: I think it's obvious. We can delete this comment.

File: features/support/world.js [hunk 4]
```diff
@@ -152,6 +151,30 @@ function ensurePipelineExists(config) {
                     case 'parallel_B2':
                         this.parallel_B2JobId = job.id;
                         break;
+                    case 'simple':
```
Comment (yuichi10) [inline features/support/world.js:154]: Where do we use these values?

Comment (esuc4se) [reply features/support/world.js:154]: it's used here.
https://github.com/sonic-screwdriver-cd/screwdriver/blob/d3e1e2bef048831705278df0a3dd0235be6e3fe5/features/trigger.feature#L72

Comment (yuichi10) [reply features/support/world.js:154]: Could you tell me a more specific place?
I still don't know which code does use these kinds of variables.
https://github.com/screwdriver-cd/screwdriver/pull/2159/files#diff-2bdd2f3b6b7a4cd5baf33e3462df7091R147

Comment (esuc4se) [reply features/support/world.js:154]: Sorry, these values are unnecessary.

File: features/step_definitions/trigger.js [hunk 5]
```diff
@@ -14,148 +15,225 @@ Before(
...
-        this.pipelineIdB = null;
         this.success_AJobId = null;
         this.fail_AJobId = null;
         this.success_BJobId = null;
         this.fail_BJobId = null;
         this.parallel_AJobId = null;
         this.parallel_B1JobId = null;
         this.parallel_B2JobId = null;
-        this.builds = null;
+
```
Comment (sugarnaoming) [inline features/step_definitions/trigger.js:25]: I don't think need this blank line.

Comment (esuc4se) [reply features/step_definitions/trigger.js:25]: thx

**Pull request merged**